exonum/src/runtime/mod.rs

```diff
@@ -329,11 +333,21 @@ impl fmt::Display for RuntimeFeature {
 ///
 /// ```text
 /// LIFE ::= initialize (GENESIS | RESUME) BLOCK* shutdown
-/// GENESIS ::= (deploy_artifact | initiate_adding_service update_service_status)* after_commit
-/// RESUME ::= (deploy_artifact | update_service_status)* on_resume
+/// GENESIS ::=
+///     deploy_artifact*
+///     (initiate_adding_service update_service_status)*
+///     after_commit
+/// RESUME ::= (deploy_artifact | update_service_status | migrate)* on_resume
 /// BLOCK* ::= PROPOSAL+ COMMIT
-/// PROPOSAL ::= before_transactions* (execute | initiate_adding_service)* after_transactions*
-/// COMMIT ::= deploy_artifact* update_service_status* after_commit
+/// PROPOSAL ::=
+///     (before_transactions CALL*)*
```

Comment:
Are you saying "before_transactions can also execute (normal) transactions"?

Reply:
Yeah, `before_transactions` and `after_transactions` handlers can execute calls to the service. I'll elaborate this below in the docs.